contracts/multisig/src/key.rs

```diff
@@ -211,37 +215,40 @@ impl KeyDeserialize for KeyType {
     }
 }
 
-const ECDSA_COMPRESSED_PUBKEY_LEN: usize = 33;
-const ECDSA_UNCOMPRESSED_PUBKEY_LEN: usize = 65;
-
-const ED25519_PUBKEY_LEN: usize = 32;
-const ED25519_SIGNATURE_LEN: usize = 64;
+// It is possible for the same public key to be represented in different formats. For example
+// an ECDSA public key can be represented as compressed or uncompressed. This function
+// validates and normalizes the public key to a single format.
+fn validate_and_normalize_public_key(
+    key_type: KeyType,
+    pub_key: HexBinary,
+) -> Result<HexBinary, ContractError> {
+    match key_type {
+        KeyType::Ecdsa => Ok(k256::PublicKey::from_sec1_bytes(pub_key.as_slice())
+            .map_err(|_| ContractError::InvalidPublicKey)?
+            .to_sec1_bytes()
+            .as_ref()
+            .into()),
+        KeyType::Ed25519 => Ok(ed25519_dalek::VerifyingKey::from_bytes(
```

Comment:
From the crate's [docs](https://docs.rs/ed25519-dalek/latest/ed25519_dalek/struct.VerifyingKey.html#method.from_bytes)
> Warning
>The caller is responsible for ensuring that the bytes passed into this method actually represent a >curve25519_dalek::curve::CompressedEdwardsY and that said compressed point is actually a point on the curve.

Looks like `from_bytes` doesn't do sufficient validation. It's also not clear to me which encoding scheme this crate is using (whether it's a custom one or another). If it's custom than it makes it harder for us to swap the ed25519 signer backend in the future. You can try investigating if there's some standard format and validation we can use. Otherwise, add a TODO and create a JIRA ticket and I'll take a look later.

Reply:
created this two Jira tasks to tackle this in a future PR:

https://axelarnetwork.atlassian.net/browse/AXE-3312
https://axelarnetwork.atlassian.net/browse/AXE-3313